gossipd/gossmap_manage.c

```diff
@@ -318,6 +318,7 @@ static void remove_channel(struct gossmap_manage *gm,
 					      GOSSIP_STORE_DYING_BIT,
 					      WIRE_NODE_ANNOUNCEMENT);
 	}
+	gossmap_remove_chan(gossmap, chan);
```

Comment:
I don't think this is right.  We should never call gossmap_remove_chan manually like this.

We already append the tombstone marker above (WIRE_GOSSIP_STORE_DELETE_CHAN), so the next gossmap refresh should do the delete properly.